src/can.rs

```diff
@@ -0,0 +1,357 @@
+//! CAN bus peripheral control.
+//!
+//! It is called Two-Wire Automotive Interface (TWAI) in ESP32 documentation.
+//!
+
+use crate::delay::{portMAX_DELAY, TickType};
+use crate::gpio::*;
+use core::marker::PhantomData;
+use core::time::Duration;
+use esp_idf_sys::*;
+
+/// CAN timing
+pub enum Timing {
```

Comment:
Given that this is an `enum`, we probably want to derive all the usual stuff: Default, Copy, Clone, Eq, PartialEq. See [here](https://github.com/esp-rs/esp-idf-hal/blob/master/src/adc.rs#L66) for a sample.